### ISSUE TITLE
Skmanage returning an invalid message when using an inactive (but listed) router id

### ISSUE THREAD

**Issue by fgiorgetti** [skupperproject/skupper-router#1049]
I have noticed one issue with skmanage in the following situation. Suppose I have two linked sites: sk2 -> sk1.

If I run skmanage on sk2 to query the router nodes, I get this:
```json
[
  {
    "id": "sk2-skupper-router-5787ff9d4f-sdnjc",
    "protocolVersion": 1,
    "instance": 1680616582,
    "linkState": [
      "sk1-skupper-router-56bbc5dd7c-svp9j"
    ],
    "nextHop": "(self)",
    "validOrigins": [],
    "address": "amqp:/_topo/0/sk2-skupper-router-5787ff9d4f-sdnjc",
    "lastTopoChange": 1680628061,
    "index": 0,
    "name": "router.node/sk2-skupper-router-5787ff9d4f-sdnjc",
    "identity": "router.node/sk2-skupper-router-5787ff9d4f-sdnjc",
    "type": "io.skupper.router.router.node"
  },
  {
    "id": "sk1-skupper-router-56bbc5dd7c-svp9j",
    "protocolVersion": 1,
    "instance": 1680625753,
    "linkState": [
      "sk2-skupper-router-5787ff9d4f-sdnjc"
    ],
    "validOrigins": [],
    "address": "amqp:/_topo/0/sk1-skupper-router-56bbc5dd7c-svp9j",
    "routerLink": 0,
    "cost": 1,
    "index": 1,
    "name": "router.node/sk1-skupper-router-56bbc5dd7c-svp9j",
    "identity": "router.node/sk1-skupper-router-56bbc5dd7c-svp9j",
    "type": "io.skupper.router.router.node"
  }
]
```

Then supposing I am going to iterate through the nodes and perform this same query again, but this time passing the router id, in this case the router id for sk1, using:

`skmanage query --type io.skupper.router.router.node -r sk1-skupper-router-56bbc5dd7c-svp9j`

It works fine, see:

```json
[
  {
    "id": "sk1-skupper-router-56bbc5dd7c-svp9j",
    "protocolVersion": 1,
    "instance": 1680625753,
    "linkState": [
      "sk2-skupper-router-5787ff9d4f-sdnjc"
    ],
    "nextHop": "(self)",
    "validOrigins": [],
    "address": "amqp:/_topo/0/sk1-skupper-router-56bbc5dd7c-svp9j",
    "lastTopoChange": 1680628103,
    "index": 0,
    "name": "router.node/sk1-skupper-router-56bbc5dd7c-svp9j",
    "identity": "router.node/sk1-skupper-router-56bbc5dd7c-svp9j",
    "type": "io.skupper.router.router.node"
  },
  {
    "id": "sk2-skupper-router-5787ff9d4f-sdnjc",
    "protocolVersion": 1,
    "instance": 1680616582,
    "linkState": [
      "sk1-skupper-router-56bbc5dd7c-svp9j"
    ],
    "validOrigins": [],
    "address": "amqp:/_topo/0/sk2-skupper-router-5787ff9d4f-sdnjc",
    "routerLink": 0,
    "cost": 1,
    "index": 2,
    "name": "router.node/sk2-skupper-router-5787ff9d4f-sdnjc",
    "identity": "router.node/sk2-skupper-router-5787ff9d4f-sdnjc",
    "type": "io.skupper.router.router.node"
  }
]
```

Then if I remove the link from sk2 to sk1 and repeat both operations again, I am seeing an invalid response.

After doing it, the first query (against sk2) still returns sk1 as a router node, if I try to query again using `-r sk1-skupper-router-56bbc5dd7c-svp9j`, I am seeing:

```
SendException: RELEASED
```

**Comment by jiridanek** [comment]
* https://github.com/skupperproject/skupper/pull/1087

**Comment by ganeshmurthy** [comment]
The question is, should we improve skmanage to always return a json. We are not planning to do that at least for the time being. 
I am closing this as "Not planned" but we can reopen this if requirements change in the future.

**Comment by grs** [comment]
The question in my mind is whether `SendException: RELEASED` is the best we can do for an error message if an incorrect router-id is specified. Not an urgent or serious issue, but helpfulness of error messages is a key part of usability.

**Comment by jiridanek** [comment]
> The question in my mind is whether `SendException: RELEASED` is the best we can do for an error message if an incorrect router-id is specified.

That's what the router network does when it tries to deliver the management message to the targeted router, but since the router is missing and there is nowhere to deliver the message, it gives back the RELEASED disposition?

If so, then the response for querying a completely bogus router id should be (currently) something still different, I'd guess.

> The question is, should we improve skmanage to always return a json. We are not planning to do that at least for the time being. I am closing this as "Not planned" but we can reopen this if requirements change in the future.

If the idea is that skmanage is to have machine-friendly output, then probably yes. I am honestly surprised that skupper runs skmanage, I thought it sends its own AMQP management queries directly, with the go amqp lib.

**Comment by grs** [comment]
> That's what the router network does when it tries to deliver the management message to the targeted router, but since the router is missing and there is nowhere to deliver the message, it gives back the RELEASED disposition?
> 
> If so, then the response for querying a completely bogus router id should be (currently) something still different, I'd guess.

Good point. However in each case, it seems like the skmanage tool could interpret the error it encounters a present something more meaningful to users. In skupper it is used mostly for debugging (as you say, most cases that are able to use the management protocol directly).

To me, machine friendly output is a separate issue from helpful error messages (i.e. error that don't require a router expert to interpret them).

**Comment by fgiorgetti** [comment]
> If the idea is that skmanage is to have machine-friendly output, then probably yes. I am honestly surprised that skupper runs skmanage, I thought it sends its own AMQP management queries directly, with the go amqp lib.

It uses AMQP directly for components running in k8s (like another container).
The CLI sometime needs to use skmanage, as the AMQP port is not exposed externally.
